File: A's/cpp/59A.cpp
#include <bits/stdc++.h>

using namespace std;

int main() {
    string s; cin >> s;

    string uppercase = "QWERTYUIOPASDFGHJKLZXCVBNM";

    int n = s.size();
    int u {0}; int l {0};
    for (int i = 0; i < n; i++) {
        if (uppercase.find(s[i]) != string::npos) {  // If in uppercase
            u++;
        } else {
            l++;
        }
    }

    if (u > l) {  // More uppercase letters
        transform(s.begin(), s.end(), s.begin(), ::toupper);
    } else {
        transform(s.begin(), s.end(), s.begin(), ::tolower);
    }

    cout << s << endl;
    
    return 0;
}

File: A's/py/59A.py
def main() -> None:
    s: str = str(input())
    
    uppercase: str = "QWERTYUIOPASDFGHJKLZXCVBNM"
    
    u: int = 0
    l: int = 0

    for c in s:
        if c in uppercase: u += 1
        else: l += 1

    if u > l:  # More uppercase letters
        s = s.upper()
    else: s = s.lower()

    print(s)

if __name__ == "__main__":
    main()